src/disc/Albany_AbstractDiscretization.hpp

```diff
@@ -389,6 +392,16 @@ class AbstractDiscretization
       const bool               overlapped = false,
       const bool               force_write_solution = false) = 0; 
 
+  // Check if mesh adaptation is needed, and if so what kind (topological or just mesh-movement)
+  virtual Teuchos::RCP<AdaptationData>
```

Comment:
I'm thinking I may change this. Namely, this may just return an AdaptationType enum. The reason is that instead of doing
```c++
auto data = disc->checkForAdaptation(x,xdot,xdotdot,dxdp);
if (data->type!=None) {
  disc->adapt(data);
}
```
We could just do
```c++
auto type = disc->checkForAdaptation(x,xdot,xdotdot,dxdp);
if (type!=None) {
  disc->adapt(x,xdot,xdotdot,dxdp);
}
```
without forcing the disc to stuff the OLD x/xdot/xdotdot/dxdp inside the returned type (the caller site already has them anyways).